### PR TITLE
Add Potion NamespacedKey Patches

### PR DESCRIPTION
**Pull request by DaRacci**
New PR into 1.18.1, no changes since last PR

### CODE REVIEW

**Review by encode42 — APPROVED** [review]

**Comment by Camotoy** [comment]
I don't understand the purpose of this PR. Doesn't the persistent data container exist for this reason? Is there something here that existing functionality doesn't provide? Besides that, have you tested if this data stays should the user switch from Purpur to another downstream server build? Also, why specifically for this one entity type? In my opinion, this seems like a really, really niche feature.

**Comment by Camotoy** [comment]
My apologies; this is for actual effects, not any sort of entity. Though the point about the data persisting between server forks *should* still persist.

**Comment by DaRacci** [comment]
> My apologies; this is for actual effects, not any sort of entity. Though the point about the data persisting between server forks _should_ still persist.

I haven't done any form of testing between forks however due to modifying the serializer the data wont persist through forks as it won't load the key and would override it once saved again

File: patches/api/0043-Potion-NamespacedKey.patch [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+@@ -242,9 +298,9 @@ public class PotionEffect implements ConfigurationSerializable {
+      * @return color of this potion's particles. May be null if the potion has no particles or defined color.
+      * @deprecated color is not part of potion effects
+      */
+-     @Deprecated
+-     @Nullable
+-     @Contract("-> null")
++    @Deprecated
++    @Nullable
++    @Contract("-> null")
```
Comment (encode42) [inline patches/api/0043-Potion-NamespacedKey.patch]: Unmodify these lines so we don't run into upstream conflicts

**Review by encode42 — CHANGES_REQUESTED** [review]

**Comment by encode42** [comment]
> Though the point about the data persisting between server forks should still persist.

Due to the nature of NBT's implementation, all unused data will be wiped. This applies to Paper and all other Purpur patches that modify NBT data. The general census is that when you switch forks, you'll lose all of the fork's features.

Because of this, I believe it's fine if the data does not persist between forks.

File: patches/api/0043-Potion-NamespacedKey.patch (hunk 1 above)
Comment (DaRacci) [reply patches/api/0043-Potion-NamespacedKey.patch]: I cannot count how many times ive fixed this now, For some reason intellij keeps formatting them and i dont notice until i get told about it like this

**Review by BillyGalbreath — APPROVED** [review]

**Pull request merged**